goal_src/levels/beach/sculptor.gc

```diff
@@ -5,3 +5,1314 @@
+      (-> obj ambient)
+      (let ((t9-2 (method-of-type ambient-control dummy-11)))
+       "SCU-LO01"
+       (-> obj root trans)
+       (t9-2)
+       )
+      )
+     ((< 0.71428573 f0-2)
+      (-> obj ambient)
+      (let ((t9-3 (method-of-type ambient-control dummy-11)))
+       "SCU-AM05"
+       (-> obj root trans)
+       (t9-3)
+       )
+      )
+     ((< 0.5714286 f0-2)
+      (-> obj ambient)
+      (let ((t9-4 (method-of-type ambient-control dummy-11)))
+       "SCU-AM06"
+       (-> obj root trans)
+       (t9-4)
+       )
+      )
+     ((< 0.42857143 f0-2)
+      (-> obj ambient)
+      (let ((t9-5 (method-of-type ambient-control dummy-11)))
+       "SCU-AM03"
+       (-> obj root trans)
+       (t9-5)
+       )
+      )
+     ((< 0.2857143 f0-2)
+      (-> obj ambient)
+      (let ((t9-6 (method-of-type ambient-control dummy-11)))
+       "SCU-AM04"
+       (-> obj root trans)
+       (t9-6)
+       )
+      )
+     ((< 0.14285715 f0-2)
+      (-> obj ambient)
+      (let ((t9-7 (method-of-type ambient-control dummy-11)))
+       "SCU-AM01"
+       (-> obj root trans)
+       (t9-7)
+       )
+      )
+     (else
+      (-> obj ambient)
+      (let ((t9-8 (method-of-type ambient-control dummy-11)))
+       "SCU-AM02"
+       (-> obj root trans)
+       (t9-8)
+       )
+      )
+     )
+    )
+   )
+  (none)
+  )
+
+;; failed to figure out what this is:
+(defstate idle (sculptor)
+  :virtual #t
+  :code
+  (behavior ()
+   (when (!= (if (> (-> self skel active-channels) 0)
+              (-> self skel root-channel 0 frame-group)
+              )
+          (get-3rd-art-elem self)
+          )
+    (ja-channel-push! 1 60)
+    (let ((gp-0 (-> self skel root-channel 0)))
+     (set!
+      (-> gp-0 frame-group)
+      (the-as art-joint-anim (get-3rd-art-elem self))
+      )
+     )
+    )
+   (let ((a0-8 (-> self skel root-channel 0)))
+    (set!
+     (-> a0-8 param 0)
+     (the float (+ (-> a0-8 frame-group data 0 length) -1))
+     )
+    (set! (-> a0-8 param 1) 1.0)
+    (joint-control-channel-group!
+     a0-8
+     (the-as art-joint-anim #f)
+     num-func-seek!
+     )
+    )
+   (while (not (ja-done? 0))
+    (suspend)
+    (ja-eval)
+    )
+   (let ((v1-16 (current-status (-> self tasks))))
+    (when (or (zero? v1-16) (= v1-16 7))
+     (let ((a0-12 (-> self skel root-channel 0)))
+      (set!
+       (-> a0-12 frame-group)
+       (the-as art-joint-anim (-> self draw art-group data 12))
+       )
+      (set!
+       (-> a0-12 param 0)
+       (the
+        float
+        (+
+         (->
+          (the-as art-joint-anim (-> self draw art-group data 12))
+          data
+          0
+          length
+          )
+         -1
+         )
+        )
+       )
+      (set! (-> a0-12 param 1) 1.0)
+      (set! (-> a0-12 frame-num) 0.0)
+      (joint-control-channel-group!
+       a0-12
+       (the-as art-joint-anim (-> self draw art-group data 12))
+       num-func-seek!
+       )
+      )
+     (until (begin
```

Comment:
hopefully if you run again with the changes in https://github.com/water111/jak-project/pull/768, this loop will look a lot nicer